Applications/ApplicationsLib/ProjectData.cpp

```diff
@@ -186,6 +187,33 @@ void ProjectData::buildProcesses()
                     *_mesh_vec[0], *nl_slv, std::move(time_disc),
                     std::move(conv_crit), _process_variables, _parameters, pc));
         }
+        else if (type == "SMALL_DEFORMATION")
+        {
+            switch (pc.getConfigParameter<int>("dimension"))
```

Comment:
My current understanding is that we would only need that for structural elements (plates, shells, membranes, etc.). These elements require special element formulations resting on physical input, i.e. postulated deformation modes, that would have to be implemented separately. As we're dealing only with 3D continuum and 2D plane strain continuum elements in formulations, where the deformation takes place only in the mesh domain (e.g. in-plane), I'd say that indeed we can derive the dimension from the mesh.